Source/Core/Core/IOS/Network/IP/Top.cpp

```diff
@@ -155,802 +155,844 @@ IPCCommandResult NetIPTop::IOCtl(const IOCtlRequest& request)
-             request.request == IOCTL_SO_ICMPCLOSE ? "IOCTL_SO_ICMPCLOSE" : "IOCTL_SO_CLOSE", fd,
-             return_value);
-    break;
-  }
+    return Close(request);
   case IOCTL_SO_ACCEPT:
   case IOCTL_SO_BIND:
   case IOCTL_SO_CONNECT:
   case IOCTL_SO_FCNTL:
-  {
-    u32 fd = Memory::Read_U32(request.buffer_in);
-    WiiSockMan& sm = WiiSockMan::GetInstance();
-    sm.DoSock(fd, request, static_cast<NET_IOCTL>(request.request));
-    return GetNoReply();
-  }
-  /////////////////////////////////////////////////////////////
-  //                  TODO: Tidy all below                   //
-  /////////////////////////////////////////////////////////////
+    return DoSock(request);
   case IOCTL_SO_SHUTDOWN:
-  {
-    request.Log(GetDeviceName(), LogTypes::IOS_WC24);
-
-    u32 fd = Memory::Read_U32(request.buffer_in);
-    u32 how = Memory::Read_U32(request.buffer_in + 4);
-    int ret = shutdown(fd, how);
-    return_value = WiiSockMan::GetNetErrorCode(ret, "SO_SHUTDOWN", false);
+    return Shutdown(request);
+  case IOCTL_SO_LISTEN:
+    return Listen(request);
+  case IOCTL_SO_GETSOCKOPT:
+    return GetSockOpt(request);
+  case IOCTL_SO_SETSOCKOPT:
+    return SetSockOpt(request);
+  case IOCTL_SO_GETSOCKNAME:
+    return GetSockName(request);
+  case IOCTL_SO_GETPEERNAME:
+    return GetPeerName(request);
+  case IOCTL_SO_GETHOSTID:
+    return GetHostID(request);
+  case IOCTL_SO_INETATON:
+    return INetAToN(request);
+  case IOCTL_SO_INETPTON:
+    return INetPToN(request);
+  case IOCTL_SO_INETNTOP:
+    return INetNToP(request);
+  case IOCTL_SO_POLL:
+    return Poll(request);
+  case IOCTL_SO_GETHOSTBYNAME:
+    return GetHostByName(request);
+  case IOCTL_SO_ICMPCANCEL:
+    return ICMPCancel(request);
+  default:
+    request.DumpUnknown(GetDeviceName(), LogTypes::IOS_NET);
     break;
   }
-  case IOCTL_SO_LISTEN:
+
+  return GetDefaultReply(0);
+}
+
+IPCCommandResult NetIPTop::IOCtlV(const IOCtlVRequest& request)
+{
+  switch (request.request)
   {
-    u32 fd = Memory::Read_U32(request.buffer_in);
-    u32 BACKLOG = Memory::Read_U32(request.buffer_in + 0x04);
-    u32 ret = listen(fd, BACKLOG);
-    return_value = WiiSockMan::GetNetErrorCode(ret, "SO_LISTEN", false);
-    request.Log(GetDeviceName(), LogTypes::IOS_WC24);
+  case IOCTLV_SO_GETINTERFACEOPT:
+    return GetInterfaceOpt(request);
+  case IOCTLV_SO_SENDTO:
+    return SendTo(request);
+  case IOCTLV_SO_RECVFROM:
+    return RecvFrom(request);
+  case IOCTLV_SO_GETADDRINFO:
+    return GetAddressInfo(request);
+  case IOCTLV_SO_ICMPPING:
+    return ICMPPing(request);
+  default:
+    request.DumpUnknown(GetDeviceName(), LogTypes::IOS_NET);
     break;
   }
-  case IOCTL_SO_GETSOCKOPT:
-  {
-    u32 fd = Memory::Read_U32(request.buffer_out);
-    u32 level = Memory::Read_U32(request.buffer_out + 4);
-    u32 optname = Memory::Read_U32(request.buffer_out + 8);
 
-    request.Log(GetDeviceName(), LogTypes::IOS_WC24);
+  return GetDefaultReply(0);
```

Comment:
IPC_SUCCESS
As a side note (probably for myself mainly :P), I doubt we should return success here. We should probably return IPC_EINVAL instead for unknown ioctlvs.